README.md

```diff
@@ -41,6 +41,14 @@ Provides the following hooks:
 3. Run `pre-commit install` to add pre-commit git hooks. You can also run
    `pre-commit run --all-files`, which is useful as part of your tests.
 
-Note that currently it's only possible to use the Puppet 3 parser for
-validation. The Puppet 4 parser has many changes, and it's planned to have a
-separate branch for Puppet 4 hooks in the future.
+To use the future parser with validation, just pass it as an argument
+to the `puppet-validate` hook:
+
+    -   id: puppet-validate
+        args: [--parser=future]
+
+This option will only work for Puppet 3. For Puppet 4, just use a
+updated version of `puppet` to run these hooks and remove the argument.
```

Comment:
I don't think this is actually possible since these hooks pin puppet 3.8.1.

I think the proper thing to do is update the hooks here to use the new `additional_dependencies` support in pre-commit. This also allows overriding that version if desired.